File: Android/core/core-network/src/main/java/com/youkong/core/network/model/GridModels.kt
package com.youkong.core.network.model

import kotlinx.serialization.SerialName
import kotlinx.serialization.Serializable

/**
 * 宫格响应
 */
@Serializable
data class GridResponse(
    @SerialName("grid_size")
    val gridSize: Int,
    val friends: List<FriendGridItem>
)

/**
 * 宫格中的好友项
 */
@Serializable
data class FriendGridItem(
    @SerialName("user_id")
    val userId: String,
    val nickname: String,
    val avatar: String? = null,
    val emoji: String,
    val status: String,
    @SerialName("updated_at")
    val updatedAt: String,
    @SerialName("relative_time")
    val relativeTime: String,
    val city: String? = null,
    @SerialName("is_available")
    val isAvailable: Boolean = false,
    @SerialName("is_visiting")
    val isVisiting: Boolean = false,
    @SerialName("gif_url")
    val gifUrl: String? = null,
    @SerialName("giphy_query")
    val giphyQuery: String? = null,
    @SerialName("use_gif")
    val useGif: Boolean = false,
    @SerialName("needs_schedule")
    val needsSchedule: Boolean = false,
    // 像素场景
    @SerialName("scene_pose")
    val scenePose: String? = null,
    @SerialName("scene_arms")
    val sceneArms: String? = null,
    @SerialName("scene_expression")
    val sceneExpression: String? = null,
    @SerialName("scene_prop")
    val sceneProp: String? = null,
    @SerialName("scene_surface")
    val sceneSurface: String? = null,
    // Rive 动画状态
    @SerialName("rive_state")
    val riveState: String? = null,
    // AI 互动选项
    val interactions: List<InteractionOptionItem>? = null,
    // 今日互动计数
    @SerialName("interaction_count")
    val interactionCount: Int = 0,
)

/**
 * AI 互动选项
 */
@Serializable
data class InteractionOptionItem(
    val emoji: String,
    val label: String,
    @SerialName("push_text")
    val pushText: String,
)

/**
 * 发送互动请求
 */
@Serializable
data class SendInteractionRequest(
    @SerialName("receiver_id")
    val receiverId: String,
    @SerialName("action_emoji")
    val actionEmoji: String,
    @SerialName("action_label")
    val actionLabel: String,
    @SerialName("action_push_text")
    val actionPushText: String,
)

/**
 * AI 自动推测就绪详情
 */
@Serializable
data class AIReadyDetails(
    @SerialName("perm_location")
    val permLocation: Boolean = false,
    @SerialName("perm_motion")
    val permMotion: Boolean = false,
    @SerialName("perm_calendar")
    val permCalendar: Boolean = false,
    @SerialName("has_invited_friend")
    val hasInvitedFriend: Boolean = false,
    @SerialName("has_voice_schedule")
    val hasVoiceSchedule: Boolean = false,
)

File: iOS/YouKong/Domain/Entities/GridModels.swift
import Foundation

// MARK: - Grid Models

/// 宫格响应
struct GridResponse: Codable {
    let gridSize: Int
    let friends: [FriendGridItem]

    enum CodingKeys: String, CodingKey {
        case gridSize = "grid_size"
        case friends
    }
}

/// 宫格中的好友项
struct FriendGridItem: Codable, Identifiable {
    let userId: String
    let nickname: String
    let avatar: String?
    let emoji: String
    let status: String
    let updatedAt: String
    let relativeTime: String
    let city: String?
    let isAvailable: Bool?
    let isVisiting: Bool?
    let gifUrl: String?
    let giphyQuery: String?
    let useGif: Bool?
    let needsSchedule: Bool?
    // 像素场景
    let scenePose: String?
    let sceneArms: String?
    let sceneExpression: String?
    let sceneProp: String?
    let sceneSurface: String?
    // AI 互动选项
    let interactions: [InteractionOptionItem]?
    // 今日互动计数
    let interactionCount: Int?
    // 是否已对该好友当前状态 +1
    let hasPlusOned: Bool?
    // 是否是自己
    let isSelf: Bool?

    var id: String { userId }

    enum CodingKeys: String, CodingKey {
        case userId = "user_id"
        case nickname
        case avatar
        case emoji
        case status
        case updatedAt = "updated_at"
        case relativeTime = "relative_time"
        case city
        case isAvailable = "is_available"
        case isVisiting = "is_visiting"
        case gifUrl = "gif_url"
        case giphyQuery = "giphy_query"
        case useGif = "use_gif"
        case needsSchedule = "needs_schedule"
        case scenePose = "scene_pose"
        case sceneArms = "scene_arms"
        case sceneExpression = "scene_expression"
        case sceneProp = "scene_prop"
        case sceneSurface = "scene_surface"
        case interactions
        case interactionCount = "interaction_count"
        case hasPlusOned = "has_plus_oned"
        case isSelf = "is_self"
    }

    /// 转换为 PixelSceneConfig
    var pixelSceneConfig: PixelSceneConfig? {
        guard let pose = scenePose, !pose.isEmpty else { return nil }
        return PixelSceneConfig(
            pose: pose,
            arms: sceneArms ?? "down",
            expression: sceneExpression ?? "normal",
            prop: sceneProp ?? "none",
            surface: sceneSurface ?? "none"
        )
    }
}

/// AI 互动选项
struct InteractionOptionItem: Codable, Identifiable {
    let emoji: String
    let label: String
    let pushText: String

    var id: String { emoji + label }

    enum CodingKeys: String, CodingKey {
        case emoji
        case label
        case pushText = "push_text"
    }
}

/// 发送互动请求
struct SendInteractionRequest: Codable {
    let receiverId: String
    let actionEmoji: String
    let actionLabel: String
    let actionPushText: String

    enum CodingKeys: String, CodingKey {
        case receiverId = "receiver_id"
        case actionEmoji = "action_emoji"
        case actionLabel = "action_label"
        case actionPushText = "action_push_text"
    }
}
